src/base/utils/string.h

```diff
@@ -36,19 +36,45 @@
 #include <QString>
 #include <Qt>
 #include <QtContainerFwd>
+#include <QtGlobal>
+
+#ifndef Q_OS_WIN
+#define QBT_USE_QCOLLATOR
+#include <QCollator>
+#endif
 
 class QStringRef;
 
 namespace Utils::String
 {
-    QString fromDouble(double n, int precision);
+#ifndef QBT_USE_QCOLLATOR
+    int naturalCompare(const QString &left, const QString &right, Qt::CaseSensitivity caseSensitivity);
+#endif
 
-    int naturalCompare(const QString &left, const QString &right, const Qt::CaseSensitivity caseSensitivity);
     template <Qt::CaseSensitivity caseSensitivity>
-    bool naturalLessThan(const QString &left, const QString &right)
+    class NaturalCompare
```

Comment:
Is there some significant profit of making this class template? Having it as a regular class would make the header less cluttered with implementation details.

Reply:
I'm trying to have an unified interface: `NaturalCompare<>` and `NaturalLessThan<>`.
Note that C++20 also has a similar class: `std::compare_three_way`.

If you insist I can change it but I hope we don't need to revise it again in the future when the need arise.

Reply:
>I'm trying to have an unified interface: `NaturalCompare<>` and `NaturalLessThan<>`.

Sorry, don't understand, why you can't achieve it using regular classes.

Reply:
>Sorry, don't understand, why you can't achieve it using regular classes.

It can but will it be usable in the future? I mean for similar situations like interfacing with std::set.

Reply:
>I mean for similar situations like interfacing with `std::set`.

Indeed, in this case, we will need the name of the Callable.